packages/react-components/src/DocsComponents/FluentDocsDocsStory.stories.tsx

```diff
@@ -0,0 +1,168 @@
+  const sandpack = useSandpack();
+
+  React.useEffect(() => {
+    if (id) {
+      const anchor = document.getElementById(`anchor--${id}`);
+      const iframe = anchor?.getElementsByTagName('iframe')?.item(0);
+
+      // toggle iframe/inline rendering
+      if (iframe) {
+        iframe.style.display = 'none';
+
+        sandpack.listen(msg => {
+          if (msg.type === 'start') {
+            iframe.style.display = 'block';
+            setRenderInline(false);
+          }
+        });
+      }
+    }
+  }, [id]);
+
+  return (
+    <SandpackPreview
+      actionsChildren={
+        <>
+          <button
+            type="button"
+            className="sp-button"
+            style={{ padding: 'var(--sp-space-1) var(--sp-space-3)' }}
+            onClick={handleReportBug(componentName)}
+          >
+            Report bug
+          </button>
+        </>
+      }
+    >
+      <>{renderInline && <Story id={id} />}</>
+    </SandpackPreview>
+  );
+};
+
+// Most of this file is copied from Storybook's addons/docs/src/blocks/DocsStory.tsx
+export const FluentDocsDocsStory: React.FunctionComponent<DocsStoryProps & { componentName: string }> = ({
+  id,
+  name,
+  expanded = true,
+  withToolbar = false,
+  parameters = {},
+  componentName,
+}) => {
+  const { docs } = parameters;
+
+  if (!name) {
+    return null;
+  }
+
+  return (
+    <Anchor storyId={id ?? ''}>
+      {expanded && name && <Subheading>{name}</Subheading>}
+      {expanded && docs && docs.description?.story && <Description markdown={docs.description?.story} />}
+      <div className="sbdocs sbdocs-preview">
+        <SandpackProvider
+          css={{ width: 'fit-content' }}
+          template="react"
+          bundlerURL="https://sandpack-bundler.pages.dev"
+          openPaths={['/example.tsx']}
+          customSetup={{
+            environment: 'create-react-app',
+            entry: '/index.tsx',
+            main: '/example.tsx',
+            files: {
+              '/example.tsx': parameters.fullSource,
+              '/index.html': '<div id="root"></div>',
+              '/index.tsx': {
+                // use originalStoryFn because users can override the `storyName` property.
+                // We need the name of the exported function, not the actual story
+                // https://github.com/microsoft/fluentui-storybook-addons/issues/12
+                code: dedent`
+            import * as ReactDOM from 'react-dom';
+            import { FluentProvider, webLightTheme } from '@fluentui/react-components';
+            import { STORY_NAME as Example } from './example';
+            //
+            // You can edit this example in "example.tsx".
+            //
+            ReactDOM.render(
+                <FluentProvider theme={webLightTheme}>
+                    <Example />
+                </FluentProvider>,
+                document.getElementById('root'),
+            );`.replace('STORY_NAME', name.replace(new RegExp(' ', 'g'), '')),
```

Comment:
```suggestion
            );`.replace('STORY_NAME', name.trim()),
```

Should not it be just `trim`?

Reply:
We need to remove the inner whitespace too